app/Http/Controllers/MessagesController.php

```diff
@@ -95,11 +96,19 @@ public function sendMessage(Message $message)
         $contest = Contest::find($contestId);
         $contest = $this->manager->appendCampaign($contest);
 
-        $users = $this->manager->getModelUsers($competition, true);
+        if (request('test')) {
+            $user = Auth::user();
+            $user = $this->userRepository->find($user->id);
+            $user = $this->manager->appendReportback($user, null);
 
-        // Only send checkin messages to users who haven't reported back.
-        if ($message->type === 'checkin') {
-            $users = $users->where('reportback', null);
+            $users = [$user];
+        } else {
+            $users = $this->manager->getModelUsers($competition, true);
```

Comment:
Actually, don't review this yet! I realized i'm doing this twice on non-test emails now. Once, to get the users to send to and another time to generate the leaderboard. Gonna think this through a bit more but I think @weerd is pushing up some work that will allow me to pull competition activity from the competition so I don't have to generate the leaderboard manually for the email.